nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBDirectory.java

```diff
@@ -54,25 +55,20 @@
     protected boolean initialized;
 
     public MongoDBDirectory(MongoDBDirectoryDescriptor descriptor) {
-        super(descriptor);
+        super(descriptor, MongoDBReference.class);
 
-        // register the references to other directories
-        addReferences(descriptor.getInverseReferences());
-        addReferences(descriptor.getMongoDBReferences());
-
-        // cache parameterization
-        String cacheEntryName = descriptor.cacheEntryName;
-        String cacheEntryNameWithoutReferencesName = descriptor.cacheEntryWithoutReferencesName;
+        // Add specific references
+        addMongoDBReferences(descriptor.getMongoDBReferences());
```

Comment:
Why do you need this ? isn't the purpose of `super#addReferences` ?
Furthermore, maybe you'll have duplicate references.